src/State/Delay.lua

```diff
@@ -0,0 +1,75 @@
+--!strict
+
+--[[
+  Constructs a new delay object which wraps a value object
+  and delays its updates
+]]
+
+local Package = script.Parent.Parent
+local PubTypes = require(Package.PubTypes)
+local useDependency = require(Package.Dependencies.useDependency)
+local initDependency = require(Package.Dependencies.initDependency)
+local updateAll = require(Package.Dependencies.updateAll)
+
+local class = {}
+
+local CLASS_METATABLE = {__index = class}
+local WEAK_KEYS_METATABLE = {__mode = "k"}
+
+--[[
+  Returns the value of the wrapped value object
+  based on delay
+]]
+function class:get(asDependency: boolean?): any
+  if asDependency ~= false then
+    useDependency(self)
+  end
+  return self._currentValue
+end
+
+--[[
+  Called when the wrapped value object changes value.
+
+  The delay object will begin a new coroutine that will
+  be delayed in execution by the delay's duration.
+]]
+function class:update(): boolean
+  local goal = self._valueState:get(false)
+
+  task.delay(
+    self._duration,
+    function()
+      self._currentValue = goal
+      updateAll(self)
+    end
+  )
+  
+  return false
+end
+
+local function Delay<T>(valueState: PubTypes.Value<T>, delayDuration: number)
```

Comment:
Nit: I think the `valueState` type should be `PubTypes.StateObject<T>` here according to 
the declaration [here](https://github.com/Elttob/Fusion/pull/94/files#diff-a9a99be0ba45d4d8da3becc16047e212b8c236764107727fd08da4a8b7751d5cR31). If I'm not wrong I think the declaration needs to be changed [here](https://github.com/Elttob/Fusion/pull/94/files#diff-70fb4ff57b372683def292e276c20d1b6e89ebde5647fcf516218fbdf422953bR103) aswell.

And maybe there is a remnant declaration from before the scheduler was removed [here](https://github.com/Elttob/Fusion/pull/94/files#diff-70fb4ff57b372683def292e276c20d1b6e89ebde5647fcf516218fbdf422953bR106).

Reply:
Fixed 1295ed7